File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/java/com/tencent/devops/scm/provider/git/tgit/auth/TGitAuthProviderFactory.java
package com.tencent.devops.scm.provider.git.tgit.auth;

import com.tencent.devops.scm.api.pojo.auth.IScmAuth;
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider;
import java.util.Arrays;
import java.util.List;

/**
 * tgit 授权提供者创建者
 */
public class TGitAuthProviderFactory {

    private static final List<TGitAuthProviderAdapter> adapters = Arrays.asList(
            new TGitUserPassAuthProviderAdapter(),
            new TGitTokenAuthProviderAdapter()
    );

    public static TGitAuthProvider create(IScmAuth gitAuth) {
        for (TGitAuthProviderAdapter adapter : adapters) {
            if (adapter.support(gitAuth)) {
                return adapter.get(gitAuth);
            }
        }
        throw new UnsupportedOperationException(String.format("gitAuth(%s) is not support", gitAuth));
    }
}

File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/kotlin/com/tencent/devops/scm/provider/git/tgit/auth/TGitAuthProviderFactory.kt
package com.tencent.devops.scm.provider.git.tgit.auth

import com.tencent.devops.scm.api.pojo.auth.IScmAuth
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider

/**
 * TGit授权提供者工厂类
 */
object TGitAuthProviderFactory {
    // 初始化适配器列表
    private val adapters = listOf(
        TGitUserPassAuthProviderAdapter(),
        TGitTokenAuthProviderAdapter()
    )

    /**
     * 创建对应的授权提供者
     * @param gitAuth 授权信息
     * @return 授权提供者实例
     * @throws UnsupportedOperationException 当授权类型不支持时抛出
     */
    fun create(gitAuth: IScmAuth): TGitAuthProvider {
        return adapters.firstOrNull { it.support(gitAuth) }?.get(gitAuth)
            ?: throw UnsupportedOperationException("gitAuth($gitAuth) is not support")
    }
}
